components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/IdentityMgtUtil.java

```diff
@@ -0,0 +1,262 @@
+        // Extract claims that have a configured store location.
+        Iterator<Map.Entry<String, String>> claimIterator = claimMap.entrySet().iterator();
+
+        // This map will hold the claims that have a configured store location.
+        Map<String, String> userStorePersistentClaims = new HashMap<>();
+        Map<String, String> identityStorePersistentClaims = new HashMap<>();
+        String userStoreDomain = userStoreManager.getRealmConfiguration().getUserStoreProperty(UserCoreConstants
+                .RealmConfig.PROPERTY_DOMAIN_NAME);
+
+        // Retrieve tenant domain from user.
+        int tenantId;
+        try {
+            tenantId = userStoreManager.getTenantId();
+        } catch (UserStoreException e) {
+            throw new UserStoreException ("Error while retrieving tenant ID for user store: " + userStoreDomain , e);
+        }
+        String tenantDomain = IdentityTenantUtil.getTenantDomain(tenantId);
+
+
+        // Iterate through the claimMap to find claims with a custom configured store location.
+        while (claimIterator.hasNext()) {
+            Map.Entry<String, String> claim = claimIterator.next();
+            String key = claim.getKey();
+            String value = claim.getValue();
+            if (StringUtils.isEmpty(key)) {
+                continue;
+            }
+            Boolean isExcluded = isUserStoreExcludedForClaim(key, tenantDomain, userStoreDomain);
+            if (isExcluded  == null) {
+                // If the claim does not have a custom persistence configured, we skip it.
+                continue;
+            }
+
+             /*
+             If the claim has a configured custom persistent location, we remove the claim from the claimMap
+             and add it to the respective map.
+             */
+            if (!isExcluded) {
+                // If the claim isn't excluded from current user store, we add it to the userStorePersistentClaims map.
+                userStorePersistentClaims.put(key, value);
+                claimIterator.remove();
+            } else {
+                // If the claim is excluded from current user store, we add it to the identityStorePersistentClaims map.
+                identityStorePersistentClaims.put(key, value);
+                claimIterator.remove();
+            }
+        }
+        return Pair.of(userStorePersistentClaims, identityStorePersistentClaims);
+
+    }
+
+
+    /**
+     * Extract the user store persistency configured claims.
+     * This method is an overloaded version of the above method.
+     * @param claimMap       Map of claims that has been retrieved from the user store.
+     * @param claims         List of claims to be filtered.
+     * @param userStoreManager UserStoreManager instance.
+     * @return Pair of HashMaps containing user store persisted and identity db persisted claims respectively.
+     */
+    public static Pair<Map<String, String>, Map<String, String>> getCustomPersistenceEnabledClaims(
```

Comment:
Why we need this method. We can reuse the same method we have to find the two arrays of user store persistence enabled and identity store persistence enabled claims.

Based on that we can manipulate the claim map. We don't need to have that logic here.